vcd/resource_vcd_nsxt_network_dhcp.go

```diff
@@ -0,0 +1,263 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var nsxtDhcpPoolSetSchema = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"start_address": &schema.Schema{
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Start address of DHCP pool IP range",
+		},
+		"end_address": &schema.Schema{
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "End address of DHCP pool IP range",
+		},
+	},
+}
+
+func resourceVcdOpenAPIDhcp() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdOpenAPIDhcpCreate,
+		ReadContext:   resourceVcdOpenAPIDhcpRead,
+		UpdateContext: resourceVcdOpenAPIDhcpUpdate,
+		DeleteContext: resourceVcdOpenAPIDhcpDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdOpenAPIDhcpImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+
+			"org_network_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Parent Org VDC network ID",
+			},
+			"pool": &schema.Schema{
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "IP ranges used for DHCP pool allocation in the network",
+				Elem:        nsxtDhcpPoolSetSchema,
+			},
+		},
+	}
+}
+
+// resourceVcdOpenAPIDhcpCreate
```

Comment:
Why this comment? :)

Reply:
Good question :) I always start with them, but in this case there is no need to have these docs at all anyway (except for `resourceVcdOpenApiDhcpUpdate` where the explanation makes sense. Removed others.